---
title: "Introduction to R function _CCMO.dep_"
author: "Yuang Tian"
date: "2021/4/1"
output:
  pdf_document: default
  html_document: default
---

The R function _CCMO.dep_ implements an powerful method for assessing disease-gene association using case-control mother-offspring data, which improves efficiency compared to the method proposed by R package _CCMO_ (refer to the reference) by incorpating missing genotypes.

## 1 Data description

The data for each mother-offspring pair are $(Y,G^c,G^m,X)$, where $Y$ is the disease status of the offspring (should be coded as 1 for case and 0 for control), $G^c$ and $G^m$ are the genotypes of offspring and mother, and $X$ is a $p$-vector of maternal covariates. 

Suppose the targeted SNP has two alleles A and a (a is the minor allele), then genotypes can be coded either as 0 (for genotype AA), 1 (for genotype Aa), or 2 (for genotype aa).

## 2. Model description

The method assumes that the offspring genotype is conditionally independent of the maternal covariates given maternal covariates. The penetrance model is as follows:
\begin{eqnarray*}
&&\mbox{pr}(Y=1|G^m,G^c,X)\\
&=&\mbox{expit}(\beta_0+\beta_{G^m}G^m+\beta_{G^c}G^c+\beta_{X}X_o+\beta_{G^mX}G^mX_m+\beta_{G^cX}G^cX_c),
\end{eqnarray*}
where expit$(t)=e^t/(1+e^t)$,$X_o$,$X_c$, and $X_m$ are functions of the covariates $X$,  $\beta_{G^c}$ and $\beta_{G^m}$ are main genetic effects for offspring and mother, respectively, $\beta_X$ is the main maternal environmental effect, $\beta_{G^mX}$ is the maternal gene-environment interaction effect, $\beta_{G^cX}$ is the offspring gene-environment interaction effct. The maternal genotype is associated with the maternal covariates through the following double additive logit (daLOG) model:
$$\mbox{pr}(G^m=k|X)=\frac{\xi_k(\theta,F)e^{k\eta'X_{G^m}}}{\sum_l\xi_l(\theta,F)e^{l\eta'X_{G^m}}},$$
where $\eta$ is a regression parameter vector,$X_{G^m}$ is a function of $X$, $\xi_k(\theta,F)$ is the probability of $G^m=k$ given the minor allele frequency $\theta$ and fixation index parameter $F$ (more details refer to the reference). 

## 3. Usage of main function _CCMO.dep_

The usage of the function _CCMO.dep_ is as follows:

```{r,eval=FALSE}
CCMO.dep(Y,gm,gc,Xo,Xm,Xc,Xgm,f,HWE)
```

Users should specify the prevalence _f_, and make sure that the genotypes are coded as 0 (for genotype AA), 1 (for genotype Aa), or 2 (for genotype aa). Users should specify _HWE_ to choose the model they prefer to use ( _TRUE_ for model with Hardy-Weinberg equilibrium and _FALSE_ for model without Hardy-Weinberg equilibrium).

## 4. Illustration of main function _CCMO.dep_
```{r,eval=FALSE}
load(SampleData.rda)
Y <- SampleData[,1]
gm <- SampleData[,2]
gc <- SampleData[,3]
X <- SampleData[,4:5]
Xo <- X
Xm <- X[,1]
Xc <- X[,1]
Xgm <- X
f <- 0.01
fit <- CCMO.dep(Y,gm,gc,Xo,Xm,Xc,Xgm,f,TRUE)
```
The point estimation results are stored in
```{r,eval=FALSE}
fit$est
```
The estimated standard errors of the estimates are stored in
```{r,eval=FALSE}
fit$sd
```
The corresponding results by the standard logistic regression are included for comparison
```{r,eval=FALSE}
fit$est.log
fit$sd.log
```
## 5. Reference

Zhang H, Mukherjee B, Arthur V, Hu G, Hochner H, Chen J (2020). An Efficient and Computationally Robust Statistical Method for Analyzing Case-Control Mother-Offspring Pair Genetic Association Studies. *Annals of Applied Statistics* **14**: 560â€“-584.
